Assets/Scripts/CandyPool.cs

```diff
@@ -5,7 +5,8 @@
 
 public class CandyPool : MonoBehaviour
 {
-
+    private readonly object getCandyLock = new object();
```

Comment:
hmm, could you tell me why these Lock objects are needed here? Locks are used mostly in multi-threaded or asynchronous code to avoid race conditions.

Reply:
They're leftovers from one of many attempts to solve the issue caused by the multiple subscription to an event, when I thought that it could have been caused by race conditions. However, I left them in because there are async calls to the ReturnCandy method in the ````CandyAnimationController```` class (````RotateMatchingCandies```` returns ````IEnumerator```` and is launched asyncronously with a coroutine).

Reply:
Could you remove these locks and try to see if you run into any issues? 🤔 if yes then we can implement a lock logic using booleans.